text/1065-remove-ember-fetch.md

```diff
@@ -0,0 +1,141 @@
+---
+stage: accepted
+start-date: 2025-01-10T00:00:00.000Z
+release-date:
+release-versions:
+teams: # delete teams that aren't relevant
+  - cli
+  - data
+  - framework
+  - learning
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1065
+project-link:
+---
+
+<!---
+Directions for above:
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+-->
+
+# Deprecate and Remove ember-fetch 
+
+## Summary
+
+This RFC proposes removing `ember-fetch` from the blueprint for new projects, and recommends alternative, more native, ways of having "managed fetch".
+
+## Motivation
+
+the package, `ember-fetch`, does a fair bit of deceptive and incorrect behavior that is incompatible with modern JavaScript tooling, such as _being_ `ember-fetch`, yet only importing from `fetch`.
+
+## Transition Path
+
+- Remove ember-fetch from all blueprints
+- Deprecate the npm package and archieve the github repo.
+- Migrate to an alternative for "managed fetch" 
+
+### What does `ember-fetch` do?
+
+_primarily_, it wraps the native `fetch` in `waitForPromise` from `@ember/test-waiters` (aka "settled state integration").
+
+
+secondarily, but not popularly used, are a series of utilities (e.g.: for checking kinds of errors). These could be copied into projects that use them and modified to fit each project's needs. 
+
+### Using native `fetch`
+
+If you only use `ember-fetch` in route model-hooks, then the settled-state integration isn't used (or rather, you rely on the routing's settled-state integration, and `ember-fetch`'s settled-state integration is extraneous) 
+
+
+### Direct replacement
+
+
+To replace `ember-fetch`'s core-functionality using the least amount of effort involves adding a new utility in your apps:
+
+```ts
+import { waitForPromise } from '@ember/test-waiters';
+
+const nativeFetch = globalThis.fetch;
+
+export async function wrappedFetch(...args: Parameters<typeof nativeFetch>) {
+    let responsePromise = nativeFetch(...args);
+
+    waitForPromise(responsePromise);
+
+    let response = await responsePromise;
+
+    return new Proxy(response, {
+        get(target, prop, receiver) {
+            let original = Reflect.get(target, prop, receiver);
+
+            if (['json', 'text'].includes(prop)) {
```

Comment:
Taken from an app that already uses this pattern:

```js
['text', 'json', 'arrayBuffer', 'blob', 'formData']
```

Reply:
updated!